default.nix

```diff
@@ -0,0 +1,84 @@
+# Building purebred
+#
+# You should be able to simply invoke:
+#
+# $ nix-build
+#
+# or, to be explicit:
+#
+# $ nix-build default.nix
+#
+# in purebred's root directory in order to build purebred. You'll find the binary under:
+#
+# $ ls result/bin/purebred
+#
+# if the build was successful.
+#
+#
+# Choosing a different compiler than the default
+#
+# In order to choose a different compiler, invoke nix build like so (escaping
+# the quotes is needed, since we're passing a string literal):
+#
+# $ nix-build --arg compiler \"ghc442\"
+#
+#
+# Use as a development environment
+#
+# $ nix-shell default.nix
+#
+{ compiler ? "ghc822" }:
+
+let
+  config = {
+    packageOverrides = super: let self = super.pkgs; in
+    {
+      haskell = super.haskell // {
+        packages = super.haskell.packages // {
+          "${compiler}" = super.haskell.packages."${compiler}".override {
+            overrides = self: super: {
+              # Build with latest known stable version.
+              purebred = with pkgs.haskell.lib; dontHaddock (self.callPackage ./.nix-helpers/purebred.nix { });
+              purebred-email = self.callPackage ./.nix-helpers/purebred-email.nix { };
+              # Version pins since those versions were bleeding edge and not on
+              # hackage/stackage or Nix yet. Remove these once builds without
+              # these pass.
+              notmuch = self.callPackage ./.nix-helpers/hs-notmuch.nix {
+                notmuch = pkgs.notmuch;
```

Comment:
Are these two lines (notmuch, talloc) necessary?

Reply:
Yes. It specifies which dependency notmuch is coming from (hs-notmuch depends on the C-sources named notmuch). More info I got from: https://github.com/Gabriel439/haskell-nix/blob/9c72b6ecbc5e25df509dfd6ee3d5ee8b9eb21f14/project2/README.md